pet.py

```diff
@@ -0,0 +1,63 @@
+import datetime
+
+# Define class
+class pet:
+    
+    def __init__(self, name, category, subtype, birthdate = None, colors = None, has_tail = False, has_wings = False, num_of_legs = 4, last_fed = datetime.date.today(), deathdate = None):
+        self.name = name
+        self.category = category
+        self.subtype = subtype
+        self.birthdate = birthdate
+        self.colors = colors
+        self.has_tail = has_tail
+        self.has_wings = has_wings
+        self.num_of_legs = num_of_legs
+        self.last_fed = last_fed
+        self.deathdate = deathdate
+
+    def getName(self):
```

Comment:
rename this and all methods to "setName" instead of "getName"

getName should RETURN the value.

This comes from the idea of "encapsulation" in OOP (Object Oriented Programming).  The idea is that the pet class should define all functionality and changes to that functionality.  You shouldn't be able to set values beyond limits (You can't set a birthdate to a date that is after a deathdate, or a deathdate to before the birthdate).

Methods like "setBirthdate" would have checks to make sure the birthdate isn't after the deathdate, or in the future.  Methods like died, would verify the date isn't before the birthdate, or in the future (How can you predict death?).  You do not have to add in any logic or checks, this is just me trying to explain why the "set" methods exist in OOP.  

The "get" methods in OOP are to allow you to control access to data.  For instance, maybe you have variables in the class that control positions of the leg joints, when a user is accessing the pet class, they probably don't care about individual joint positions, just where the cat is.  So you wouldn't "expose" the joint position variables.  Instead you'd have getPosition() that would return perhaps the center of the cats position.

Reply:
Literally the get and set comes from the idea of "getters" and "Setters".

https://www.python-course.eu/python3_properties.php